site/templates/application.jet

```diff
@@ -40,8 +40,19 @@
     <script>
       window.addEventListener('DOMContentLoaded', function(){
         s72.cfg(function(){ return s72.i18n.load('{{lang.Code}}', '/{{lang.DefinitionFilePath}}'); });
-        s72.cfg(function(){ return s72.i18n.href.load('{{site.SiteConfig.DefaultLanguage}}', '{{lang.Code}}'); });
-        s72.boot({ config: {{ site.Config | json | raw }}, toggles: {{ site.Toggles | json | raw }}, lang: '{{ lang.Code }}' });
+        s72.cfg(function(){ return s72.i18n.href.setLanguages('{{site.SiteConfig.DefaultLanguage}}', '{{lang.Code}}'); });
+        s72.boot({
+          config: {{ site.Config | json | raw }},
+          toggles: {{ site.Toggles | json | raw }},
+          languages: [
+            {{ range k, v := site.SiteConfig.Languages }}
+              {
+                value: '{{ k != site.SiteConfig.DefaultLanguage ? k : "" }}',
+                name: '{{ i18n(lower("lang_name_" + v)) }}'
+              },
+            {{ end }}
+          ]
```

Comment:
I am not a fan of this, theres too much reliance on custom mapping here.
We have 2 options:
1. Force relish code to accept the data structure kibble exposes via `site.Languages` or `site.SiteConfig.Languages`.
2. Expose a function in kibble that returns the data you want.

Reply:
What if we updated kibble to support this structure in the `kibble.json`?

```
"languages": {  
  "es": { "code": "es_ES", "name": "Español" },
  "es-MX": { "code": "es_MX", "name": "Español (Mexican)" }
}
```

Reply:
My concern with that approach (ignoring some interesting json parsing) is it doesn;t support the case where a Language name is different when referenced by different languages. That could be a non issue but English is not called English in French.

Reply:
True, but in the scenario of an English-speaker landing on the French site, they would probably be looking for the "English" option rather than "Anglais". So question is to optimize for the visitor or the language?

Reply:
Sure, lets ride with that as a decision. @PeculiarGoat are you comfortable making kibble changes, or would you like me to look at it?

Reply:
@gscragg I think option 2 is tidier.
I could stick a function on the site model `func (site *Site) LanguageSelectorOptions() []map[string]string` to return `[{ "value": "en", "name": "English" }, {etc...}]`

@stajs Yeah I think I can do that... could change `models.Config.Languages` to a typed struct and modify the unmarshalling function to handle objects instead of strings.  Then I think it would just be some small changes in the render function.  What do you think @gscragg?

Reply:
> Yeah I think I can do that... could change models.Config.Languages to a typed struct and modify the unmarshalling function to handle objects instead of strings. 

I think if we keep the models as arrays it will be easier for any implementors, the object can have `code` `locale` (or the proper iso or ietf name for it), and `name` properties.

The code that parses kibble.json will need to handle both objects and strings, as when this goes to staging/prod will potentially be used by other templates that haven't updated.

Reply:
Okay cool I'll have a go!

Reply:
Yep good point on back compat.

On the topic of naming things, I propose we try and model on IETF BCP 47 ([the same as browsers](https://www.w3.org/International/articles/language-tags/)) where it would be the `region` subtag. Future advantages could be to not just represent countries, but well, regions.

> The region subtag in RFC 3066 took its values from the ISO 3166 country codes. These two-letter codes are still available from the new registry, but the registry also lists 3-digit UN M.49 region codes. The advantage of these codes is that they can represent more than just countries. For example, localization groups have for some time wanted to label their carefully crafted translations as Latin-American Spanish, rather than the Spanish of any particular country. With RFC 5646 this is possible; the appropriate language tag is `es-419`.

We have had one request for this exact example of Latin-American Spanish so it would be nice to think outside of specific countries and have it available as a core translation.

Other potential future niceness is using the `extlang` and `script` subtags, for places like China where many different dialects are spoken and written.

> For example, `zh` means Chinese, but it covers many Chinese dialects, often mutually incomprehensible. When `zh` is used on its own, it is usually used to mean the predominant language in the encompassed range, although this is not explicitly specified in BCP 47. For example, conventionally `zh` is considered to represent the predominant, Mandarin form of Chinese. Where absolute clarity is needed you can use `cmn` instead as long as that doesn't break interoperability, however, if you are using `zh` to represent a language which is not Mandarin, such as Hakka Chinese, you are better off using the explicit code (in that case, `hak`).

Reply:
Got a bit carried away there focusing on the `region` subtag. The whole shemozzle is variously called in the RFC `language tag`, `langtag` or `tag`. I'd be keen to support regions.

Reply:
@stajs would that mean having files like `es_419.all.json` and `zh_Hans.all.json`?  Do we want to support those right away... it might work with just a bit of regex tweaking in the language selector.

Reply:
Lets leave all that extra potential niceness until later and keep this PR as small as possible to solve immediate problem.

Reply:
at the end of the day, kibble - currently - doesnt give a shit about what those keys/codes are in terms of standards or naming conventions.

We just need to pick a name for the value that is put in urls e.g `/en/`, `/es-mx/`, or `/esp/`, and a name for the value that is used to get language files e.g. `en_EN`, `eng`, or `en-GB`.

I am happy to use the same standard for the latter that html/xml uses, which @stajs suggested.